### PR TITLE
feat: support lib.id

### PR DESCRIPTION
**Pull request by fi3ework**
## Summary

Close https://github.com/web-infra-dev/rslib/issues/382.

Support configure `lib.id` in userland.

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/core/src/config.ts [hunk 1]
```diff
@@ -1179,10 +1180,16 @@ async function composeLibRsbuildConfig(config: LibConfig, configPath: string) {
   );
 }
 
+type RsbuildConfigWithLibInfo = {
```
Comment (Timeless0911) [inline packages/core/src/config.ts:1183]: put it into `types/comfig`

File: packages/core/src/types/config/index.ts [hunk 2]
```diff
@@ -72,6 +72,16 @@ export type Redirect = {
...
 export interface LibConfig extends RsbuildConfig {
+  /**
+   * Each lib configuration has a unique identifier used to distinguish different lib configurations.
+   * By default, Rslib generates a unique identifier based on the order of lib configurations, in the format `${format}${index}`.
+   * When there is only one lib of the format, the index is empty, otherwise, it starts from 0 and increments.
+   * For example:
+   * - If only ESM and CJS formats are configured, the identifier for ESM is `esm` and for CJS is `cjs`.
+   * - If two ESM formats and one CJS format are configured, they are represented as `esm0`, `esm1`, and `cjs`.
+   * @default undefined
+   */
```
Comment (Timeless0911) [inline packages/core/src/types/config/index.ts:83]: Is it necessary to write so much detail here? Just jump to website links?

File: website/docs/en/config/lib/id.mdx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+Specify the library ID. The ID identifies the library and is useful when using the `--lib` flag to build specific libraries with a meaningful `id` in the CLI.
+
+:::tip
+
+Rslib uses Rsbuild's [environments](https://rsbuild.dev/guide/advanced/environments) feature to build multiple libraries in a single project under the hood. `lib.id` will be used as the key for the generated Rsbuild environment.
+
+:::
+
+## The default ID
```
Comment (Timeless0911) [inline website/docs/en/config/lib/id.mdx:14]:
```suggestion
## Default Value
```

File: website/docs/en/config/lib/id.mdx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    { format: 'esm' }, // id is `esm0`
+    { format: 'cjs' }, // id is `cjs`
+    { format: 'esm' }, // id is `esm1`
+    { format: 'umd' }, // id is `umd`
+    { format: 'esm' }, // id is `esm2`
+  ],
+};
+```
+
+## Configuring ID
```
Comment (Timeless0911) [inline website/docs/en/config/lib/id.mdx:32]:
```suggestion
## Customize ID
```

File: packages/core/src/types/config/index.ts (hunk 2 above)
Comment (fi3ework) [reply packages/core/src/types/config/index.ts:84]: Cleaned. We need to add link for all field in another PR, like TypeScirpt.

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**